### PR TITLE
chore: bump `swc`

### PR DESCRIPTION
**Pull request by himself65**
Fixes: https://github.com/dai-shi/waku/issues/845

### CODE REVIEW

File: packages/waku/package.json [hunk 1]
```diff
@@ -88,7 +88,7 @@
   },
   "devDependencies": {
     "@netlify/functions": "^2.8.1",
-    "@swc/cli": "^0.4.0",
+    "@swc/cli": "0.4.1-nightly.20240914",
```
Comment (dai-shi) [inline packages/waku/package.json:91]: Does it conflict if we keep using `0.4.0`?

File: packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts [hunk 2]
```diff
@@ -71,17 +71,26 @@ export ${name === 'default' ? name : `const ${name} =`} createServerReference('$
...
+    start: 0,
+    end: 0,
+  }) as swc.Span;
+
 const createIdentifier = (value: string): swc.Identifier => ({
   type: 'Identifier',
   value,
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-expect-error
+  ctxt: 0,
```
Comment (dai-shi) [inline packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts:85]: Hm, we need it in this position... So, if swc fixes the TS type, the `@ts-expect-error` will be gone?

File: packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts [hunk 3]
```diff
@@ -71,17 +71,26 @@ export ${name === 'default' ? name : `const ${name} =`} createServerReference('$
...
   optional: false,
-  span: { start: 0, end: 0, ctxt: 0 },
+  span: createEmptySpan(),
```
Comment (dai-shi) [inline packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts:87]: Just to confirm, does it work if this is `span: { start: 0, end: 0 }`? Or, does TS complain?

File: packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts (hunk 2 above)
Comment (himself65) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts:85]: yes

File: packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts (hunk 3 above)
Comment (himself65) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts:87]: span: { start: 0, end: 0 }. swc type is broken

Comment (dai-shi) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts:87]: How is it likely that swc will fix this type issue sometime soon-ish?

Comment (himself65) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-transform.ts:87]: /cc @kdy1 @magic-akari

**Comment by himself65** [comment]
well, they recommend write rust

**Comment by himself65** [comment]
not sure If I should put rust version of parser to a higher prority.

**Comment by dai-shi** [comment]
I don't think it's a priority, unless it's the bottleneck of the performance.

**Comment by dai-shi** [comment]
our final goal would be rolldown, maybe?
meanwhile, we can keep using swc from js, or migrate to esbuild??

File: packages/waku/package.json (hunk 1 above)
Comment (dai-shi) [reply packages/waku/package.json:91]: oh, it is the `latest`.

**Pull request merged**